Add support for environment variable based rack awareness (#12593)
strimzi/strimzi-kafka-operator: CHANGELOG.md

* Add support for Kafka 4.1.2
* Add support for `advertisedPortTemplate` in listener configuration to add more flexibility when configuring advertised ports.
* Remove PreferredLeaderElectionGoal from Cruise Control's default.goals list

> * Add support for `type: environment-variable` rack awareness based on environmanet variables that does nto require ClusterRoleBindings.

Reviewer: ```suggestion
* Adds support for `type: environment-variable` rack awareness based on environment variables that do not require ClusterRoleBindings.
```
